modules/apps/redirect/redirect-service/src/main/java/com/liferay/redirect/internal/provider/RedirectProviderImpl.java

```diff
@@ -152,6 +164,64 @@ protected void setRedirectPatternEntries(
 		_redirectPatternEntries = redirectPatternEntries;
 	}
 
+	private String[] _getCrawlerUserAgents() {
+		if (_crawlerUserAgents == null) {
+			return new String[0];
+		}
+
+		return _crawlerUserAgents;
+	}
+
+	private boolean _isCrawlerUserAgent(
+		String[] crawlerUserAgents, String userAgent) {
+
+		if (Validator.isNull(userAgent)) {
+			return false;
+		}
+
+		for (String crawlerUserAgent : crawlerUserAgents) {
+			if (userAgent.contains(StringUtil.toLowerCase(crawlerUserAgent))) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	private boolean _isUserAgentMatch(
+		RedirectPatternEntry redirectPatternEntry, String userAgent) {
+
+		if (Validator.isNull(redirectPatternEntry.getUserAgent()) ||
+			Validator.isNull(userAgent) ||
+			Objects.equals(
+				RedirectConstants.USER_AGENT_ALL,
+				redirectPatternEntry.getUserAgent())) {
+
+			return true;
+		}
+
+		boolean crawlerUserAgent = _isCrawlerUserAgent(
+			_getCrawlerUserAgents(), userAgent);
```

Comment:
Why not call `_getCrawlerUserAgents` from inside the `_isCrawlerUserAgent` method? I don't see the need to pass it as argument as it always uses the same value.

Reply:
I have this way but I understood you wanted as a parameter... 🤷🏻‍♀️ changing again :D